x-pack/plugins/security_solution/public/resolver/view/use_camera.ts

```diff
@@ -284,14 +284,30 @@ export function useAutoUpdatingClientRect(): [DOMRect | null, (node: Element | n
   const [rect, setRect] = useState<DOMRect | null>(null);
   // Using state as ref.current update does not trigger effect hook when reset
   const [currentNode, setCurrentNode] = useState<Element | null>(null);
-
+  const [currentY, setCurrentY] = useState<number>(0);
+  const [currentX, setCurrentX] = useState<number>(0);
   const ref = useCallback((node: Element | null) => {
     setCurrentNode(node);
     if (node !== null) {
       setRect(node.getBoundingClientRect());
     }
   }, []);
   const { ResizeObserver } = useContext(SideEffectContext);
+  const handleScroll = useCallback(() => {
+    if (currentX !== window.scrollX || currentY !== window.scrollY) {
+      if (currentNode !== null) {
+        setRect(currentNode.getBoundingClientRect());
+        setCurrentY(window.scrollY);
```

Comment:
do you need to use state for `currentY` and `currentX`? If they change but `setRect` doesn't, there shouldn't be any need to render right?
how about making it like:

```
// assuming currentY and currentX are now refs
currentY.current = window.scrollY
currentX.current = window.scrollX
setRect(currentNode.getBoundingClientRect());
```